### PR TITLE
View Enclave attestation report

### PR DESCRIPTION
**Pull request by itstauq**
## Description
Closes OpenMined/Heartbeat#1607

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

File: packages/syft/src/syft/service/attestation/utils.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+    Returns:
+        Result[Ok[dict], Err[str]]: A Result object containing the payload of the verified token if successful,
+                                    or an Err object with an error message if the verification fails.
+    """
+    try:
+        # Attempt to retrieve 'jku' from the unverified header to determine the JWKS URL
+        unverified_header = jwt.get_unverified_header(token)
+        jwks_url = unverified_header.get("jku", "")
+        token_type = "cpu" if jwks_url else "gpu"
```
Comment (rasswanth-s) [inline packages/syft/src/syft/service/attestation/utils.py]: Here probably , we would need a reference, to Azure Attestation Key server, like we do nvidia, to verify if it was actually give by an Azure attestation Service.

But we could do that, when we are about to merge the PR, but this is good in the short term.

File: packages/syft/src/syft/service/project/project.py [hunk 2]
```diff
@@ -501,6 +505,45 @@ def orchestrate_enclave_execution(self) -> Any:
...
+
+        def display_report(_: widgets.Button) -> None:
+            with output:
+                output.clear_output()
+                display(JSON(report.ok()))
+
+        print("✅ Attestation report verified successfully.", flush=True)
+        button = widgets.Button(description="View full report")
+        button.on_click(display_report)
+        display(button)
```
Comment (rasswanth-s) [inline packages/syft/src/syft/service/project/project.py:563]: Nice ❤️

File: packages/syft/src/syft/service/attestation/utils.py (hunk 1 above)
Comment (itstauq) [reply packages/syft/src/syft/service/attestation/utils.py]: Added a TODO for this

File: packages/syft/src/syft/service/project/project.py [hunk 3]
```diff
@@ -501,6 +505,45 @@ def orchestrate_enclave_execution(self) -> Any:
         self.request_asset_transfer()
         return self.request_execution()
 
+    def view_attestation_report(self, return_report: bool = False) -> dict | None:
```
Comment (rasswanth-s) [inline packages/syft/src/syft/service/project/project.py]: MAYBE: shall we include a field that also allows us to get gpu attestation, or shall we keep gpu attestation to a follow up PR?

Comment (itstauq) [reply packages/syft/src/syft/service/project/project.py]: Added it in this PR

File: packages/syft/src/syft/service/attestation/utils.py (hunk 1 above)
Comment (itstauq) [reply packages/syft/src/syft/service/attestation/utils.py]: Resolved the TODO in this PR itself

**Review by rasswanth-s — APPROVED** [review]
Well Done

**Pull request merged**